File: app/src/main/java/com/example/workoutapp/ViewGenericWorkout.java
package com.example.workoutapp;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;


import android.content.Context;
import android.widget.TextView;
import android.widget.Toast;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
public class ViewGenericWorkout extends AppCompatActivity{

    public Button back, use;
    static String title, content;

    private TextView showTitle, showContent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.view_generic_workout);

        title = getIntent().getStringExtra("TITLE");
        content = getIntent().getStringExtra("CONTENT");

        back = findViewById(R.id.back);
        use = findViewById(R.id.use);

        showTitle = findViewById(R.id.title);
        showContent = findViewById(R.id.content);

        showTitle.setText(title);
        showContent.setText(content);

        back.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(ViewGenericWorkout.this, ViewGenericWorkouts.class);
                startActivity(intent);
            }
        });

        use.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent useWorkout = new Intent(ViewGenericWorkout.this, NewWorkout.class);
                useWorkout.putExtra("TITLE", title);
                useWorkout.putExtra("CONTENT", content);
                startActivity(useWorkout);
            }
        });




    }


}

File: app/src/main/java/com/example/workoutapp/ViewGenericWorkout.kt
package com.example.workoutapp

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.workoutapp.NewWorkout

class ViewGenericWorkout : AppCompatActivity() {
    var back: Button? = null
    var use: Button? = null
    private var showTitle: TextView? = null
    private var showContent: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.view_generic_workout)
        Companion.title = intent.getStringExtra("TITLE")
        content = intent.getStringExtra("CONTENT")
        back = findViewById(R.id.back)
        use = findViewById(R.id.use)
        showTitle = findViewById(R.id.title)
        showContent = findViewById(R.id.content)
        showTitle?.setText(Companion.title)
        showContent?.setText(content)
        back?.setOnClickListener(View.OnClickListener {
            val intent = Intent(this@ViewGenericWorkout, ViewGenericWorkouts::class.java)
            startActivity(intent)
        })
        use?.setOnClickListener(View.OnClickListener {
            val useWorkout = Intent(this@ViewGenericWorkout, NewWorkout::class.java)
            useWorkout.putExtra("TITLE", Companion.title)
            useWorkout.putExtra("CONTENT", content)
            startActivity(useWorkout)
        })
    }

    companion object {
        var title: String? = null
        var content: String? = null
    }
}
